DataManagementSystem/Agent/RequestOperations/ReplicateAndRegister.py

```diff
@@ -59,10 +57,16 @@ def filterReplicas( opFile, logger = None, dataManager = None, seCache = None ):
 
   replicas = replicas["Successful"].get( opFile.LFN, {} )
 
+  # FC checksum is more likely better than opFile.Checksum
+  fcMetadata = FileCatalog().getFileMetadata( opFile.LFN )
+  fcChecksum = fcMetadata.get( 'Value', {} ).get( 'Successful', {} ).get( opFile.LFN, {} ).get( 'Checksum', '' )
+  # Replace opFile.Checksum if it doesn't match a valid FC checksum
+  if fcChecksum and ( not opFile.Checksum or not compareAdler( fcChecksum, opFile.Checksum ) ):
+    opFile.Checksum = fcChecksum
```

Comment:
Moreover, should we really just silently replace it ? It would be a very abnormal situation no ?